_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,248 @@
+
+React was built out of a desire to abstract away a web page's true view hierarchy (called the DOM) so that they could make changes to all of their views and then React would handle finding the differences between view states.
+
+
+- Composition as the core concept
+- React as diff engine
+- Singular definitions of state and props
+- Provides a singular object model for web, which aims to cut across responsibilities like 
+
+React Native
+
+- Separates React from React DOM
+- Components as Views + View Controllers
+- Show how a component tree can be mapped very easily to MVC paradigm
+- Uses flexbox for layouts, which is way simpler and less likely to be buggy than auto layout
+
+Ten minutes to try out React
+
+Don't skip this, you can do this right now.
+
+- Install node
+- Install yarn
+- `yarn create react-native`
+- Install VS Code
+- Show how to edit a component
+- Explain how to turn on HMR
+- Make them edit something again, see the changes live
+
+So what is going on?
+
+- RNP as dev server
+- Create React Native
+- JSX as HTML in JS
+- JSX is basically a pretty way to write `React.createElement`
+
+Writing JavaScript
+
+- Simple and complex, flawed but fixed
+- Tooling does a really good job of keeping really bad JS code out of your app
+- Language that allows for many types of programming
+- JS is a lot of very simple tools built on top of each other, which creates a tower of dependencies
+- Always evolving, tools to let you pick and choose what features you want
+[[[[[[[[[[[[=Some useful JS terms:
+]
+  - Destructuring
+  - x
+  - y
+
+Node
+
+- No standard library
+  - + and - of this
+- Open and chaotic ecosystem
+- Often flipped by new ideas and paradigms
+
+General overview of terms you'll be interested in WRT node:
+
+- Node
+- NPM
+- Babel
+- Sourcemaps
+- Reflux
+- Redux
+- Flow
+- TypeScript
+- Lodash (underscore)
+- Relay 
+
+Web Style Development Experiene
+
+- For the web "tooling" != IDE
+- The UNIX idea of small individual libs
+- A million options, but very few attempts to provide cohesive toolkits 
+- Last 2-3 years has brought great strides forwards in the community
+  - Typed JS with inference
+  - Tooling like nuclide / vscode
+  - Safe dependency management in Yarn (and now in npm)
+  
+
+Tooling
+
+- Nuclide is good, but not good enough
+- Flow is good, but editor support is not good enough
+- We use TypeScript but it is a bit of a battle **today**
+- TypeScript + VS Code is basically Xcode level of quality, just less polished but more reliable and open source.
+- Newest release of RN includes some of our work on making 
+- node community is great at automation: linters and formatters work reliably and inside your editor
+- Debugging is tricky, but feasible. O2.
+
+Testing
+
+- Testing on native is a nightmare
+- Apple's tooling for tests has always been bad
```

Comment:
I think we should specify "always been bad in comparison to tooling for the web"